lib/src/controls/surfaces/commandbar.dart

```diff
@@ -0,0 +1,552 @@
+        w = SingleChildScrollView(
+          scrollDirection: Axis.horizontal,
+          physics: const NeverScrollableScrollPhysics(),
+          child: Row(
+            mainAxisAlignment: widget.mainAxisAlignment,
+            crossAxisAlignment: widget.crossAxisAlignment,
+            children: [
+              ...builtItems,
+              if (overflowWidget != null) overflowWidget,
+            ],
+          ),
+        );
+        break;
+    }
+    if (widget._isExpanded) {
+      w = Row(children: [Expanded(child: w)]);
+    }
+    return w;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.compactBreakpointWidth == null) {
+      return _buildForPrimaryMode(context, CommandBarItemDisplayMode.inPrimary);
+    } else {
+      return LayoutBuilder(
+        builder: (context, constraints) {
+          if (constraints.maxWidth > widget.compactBreakpointWidth!) {
+            return _buildForPrimaryMode(
+                context, CommandBarItemDisplayMode.inPrimary);
+          } else {
+            return _buildForPrimaryMode(
+                context, CommandBarItemDisplayMode.inPrimaryCompact);
+          }
+        },
+      );
+    }
+  }
+}
+
+/// When a [CommandBarItem] is being built, indicates the visual context
+/// in which the item is being built.
+enum CommandBarItemDisplayMode {
+  /// The item is displayed in the horizontal area (primary command area)
+  /// of the command bar.
+  ///
+  /// The item should be rendered by wrapping content in a
+  /// [CommandBarItemInPrimary] widget.
+  inPrimary,
+
+  /// The item is displayed in the horizontal area (primary command area)
+  /// of the command bar, but it is requested that the item take up less
+  /// horizontal space so that more items may fit without overflow.
+  ///
+  /// The item should be rendered by wrapping content in a
+  /// [CommandBarItemInPrimary] widget.
+  inPrimaryCompact,
+
+  /// The item is displayed within the secondary command area (within a
+  /// Flyout as a drop down of the "more" button).
+  ///
+  /// Normally you would want to render an item in this visual context as a
+  /// [TappableListTile].
+  inSecondary,
+}
+
+/// Signature of function that can customize the widget returned by
+/// a CommandBarItem built in the given display mode. Can be useful to
+/// wrap the widget in a [Tooltip] etc.
+typedef CommandBarItemWidgetBuilder = Widget Function(
+    BuildContext context, CommandBarItemDisplayMode displayMode, Widget w);
+
+/// An individual control displayed within a [CommandBar]. This widget ensures
+/// that the child widget has the proper margin so the item has the proper
+/// minimum height and width expected of a control within a [CommandBar].
+abstract class CommandBarItem with Diagnosticable {
+  final Key? key;
+
+  /// Specify to allow customization of the built widget for a given
+  /// display mode. For example, to wrap a widget in a [Tooltip].
+  final CommandBarItemWidgetBuilder? widgetBuilder;
```

Comment:
I don't like this. The dev can extend a `CommandBarItem` and do it on the `buildWidget` method.

Something like this can be seen on https://github.com/bdlukaa/fluent_ui/blob/master/example/lib/main.dart#L300-L339